src/interpreter/naive_interpreter.nit

```diff
@@ -2221,25 +2277,33 @@ redef class ASendExpr
 	do
 		var recv = v.expr(self.n_expr)
 		if recv == null then return null
-
-		# Safe call shortcut if recv is null
+                var my_callsite = callsite
+                if recv isa CallrefInstance then
+                        my_callsite = recv.callsite
+                        # get the original receiver
+                        recv = recv.recv
```

Comment:
I don't get it. If a method invocation is done on a callref, you switch the receiver and the callsite ?

~~~nit
var r = &x.foo
r.bar # here the receiver of `bar` is `r` the callref instance. 
~~~

Reply:
Yes, inside `CallrefInstance` there's a reference to the original receiver and callsite.

Reply:
Maybe but when sending a message to the callrefinstance, do not change the message nor the receiver.

~~~nit
import functional

class Foo
	fun bar: Int do
		print 1
		return 10
	end
end

redef class Routine
	fun baz: Int do
		print 2
		return 20
	end
end

var f = new Foo
var r = &f.bar
print r
print r.baz
print r.call
~~~

With your hunk, the interpreter crashes on this program, without it, the program is functional.

Reply:
I removed it, I don't remember why I did that since intern call are already handled by the interpreter...